src/Compilers/CSharp/Test/Emit2/Semantics/CollectionExpressionTests.cs

```diff
@@ -9498,6 +9498,700 @@ static void Main()
                 Diagnostic(ErrorCode.ERR_CollectionBuilderAttributeMethodNotFound, "[1, 2, 3]").WithArguments("Create", "int", "MyCollection").WithLocation(6, 26));
         }
 
+        [Fact]
+        public void CollectionBuilder_Conversion_ImplicitReference_NonGeneric()
+        {
+            string sourceA = """
+                using System;
+                using System.Collections;
+                using System.Collections.Generic;
+                using System.Runtime.CompilerServices;
+                [CollectionBuilder(typeof(MyCollectionBuilder), nameof(MyCollectionBuilder.Create))]
+                abstract class MyCollectionBase : IEnumerable<int>
+                {
+                    private readonly List<int> _list;
+                    protected MyCollectionBase(List<int> list) { _list = list; }
+                    public IEnumerator<int> GetEnumerator() => _list.GetEnumerator();
+                    IEnumerator IEnumerable.GetEnumerator() => GetEnumerator();
+                }
+                sealed class MyCollection : MyCollectionBase
+                {
+                    public MyCollection(List<int> list) : base(list) { }
+                }
+                class MyCollectionBuilder
+                {
+                    public static MyCollection Create(ReadOnlySpan<int> items)
+                    {
+                        return new MyCollection(new List<int>(items.ToArray()));
+                    }
+                }
+                """;
+            string sourceB1 = """
+                class Program
+                {
+                    static void Main()
+                    {
+                        MyCollectionBase x = [];
+                        x.Report();
+                        MyCollectionBase y = [1, 2, 3];
+                        y.Report();
+                    }
+                }
+                """;
+            CompileAndVerify(
+                new[] { sourceA, sourceB1, s_collectionExtensions },
+                targetFramework: TargetFramework.Net80,
+                verify: Verification.FailsPEVerify,
+                expectedOutput: IncludeExpectedOutput("[], [1, 2, 3], "));
+
+            string sourceB2 = """
+                class Program
+                {
+                    static void Main()
+                    {
+                        MyCollection x = [];
+                        MyCollection y = [1, 2, 3];
+                    }
+                }
+                """;
+            var comp = CreateCompilation(
+                new[] { sourceA, sourceB2 },
+                targetFramework: TargetFramework.Net80);
+            comp.VerifyEmitDiagnostics(
+                // 1.cs(5,26): error CS7036: There is no argument given that corresponds to the required parameter 'list' of 'MyCollection.MyCollection(List<int>)'
```

Comment:
Do you think it's worth testing a variant of `CollectionBuilderAttribute` that is marked `Inherited = true`?

Reply:
There is an existing test, `CollectionBuilder_InheritedAttributeOnBaseCollection`. Let me know if that doesn't cover the expected scenario.

Reply:
Thanks, that covers it for me.